packages/block-library/src/query/edit/index.js

```diff
@@ -75,4 +76,17 @@ export default function QueryEdit( {
 	);
 }
 
+const QueryEdit = ( props ) => {
+	const { clientId } = props;
+	const hasInnerBlocks = useSelect(
```

Comment:
I'm surprised it works out of the box when replicated from Columns block. Let's see if this is something that could be further abstracted as proposed in https://github.com/WordPress/gutenberg/issues/20582.

Reply:
Yes, I agree with that and saw that some logic might be extracted, but I think it's better for a follow up.

Another thing I wanted to wait before looking into this, is that for Query the `Placeholder` will be probably be a `steps` init block procedure. At first select some `'layout`(still block variation technically) of the block and then a second step to select some other block options. So I'd like how this could work and be abstracted for other blocks as well.